flair/models/biomedical_entity_linking.py

```diff
@@ -0,0 +1,988 @@
+            )
+            sparse_weight_path = cached_download(
+                url=sparse_weight_url,
+                cache_dir=flair.cache_root / "models" / model_name_or_path,
+            )
+
+        self.sparse_weight = torch.load(sparse_weight_path)
+
+        return self.sparse_weight
+
+    def get_sparse_weight(self) -> torch.Tensor:
+        assert self.sparse_weight is not None
+
+        return self.sparse_weight
+
+    def embed_sparse(
+        self, names: list, show_progress: bool = False, batch_size: int = 1024
+    ) -> np.ndarray:
+        """
+        Embedding data into sparse representations
+        :param names np.array: An array of names
+        :param show_progress bool: Whether to show a progress bar
+        :param batch_size int: The batch size for embedding
+        :returns sparse_embeds np.array: A list of sparse embeddings
+        """
+        sparse_embeds = []
+
+        if show_progress:
+            iterations = tqdm(
+                range(0, len(names), batch_size),
+                desc="Calculating sparse embeddings for dictionary",
+            )
+        else:
+            iterations = range(0, len(names), batch_size)
+
+        for start in iterations:
+            end = min(start + batch_size, len(names))
+            batch = names[start:end]
+            batch_sparse_embeds = self.sparse_encoder(batch)
+            batch_sparse_embeds = batch_sparse_embeds.numpy()
+            sparse_embeds.append(batch_sparse_embeds)
+        sparse_embeds = np.concatenate(sparse_embeds, axis=0)
+
+        return sparse_embeds
+
+    def embed_dense(
+        self,
+        names: np.ndarray,
+        show_progress: bool = False,
+        batch_size: int = 2048,
+    ) -> np.ndarray:
+        """
+        Embedding data into dense representations for SapBert
+        :param names: np.array of names
+        :param show_progress: bool to toggle progress bar
+        :param batch_size: batch size
+        :return dense_embeds: list of dense embeddings of the names
+        """
+        self.encoder.eval()  # prevent dropout
+
+        dense_embeds = []
+
+        with torch.no_grad():
+            if show_progress:
+                iterations = tqdm(
+                    range(0, len(names), batch_size),
+                    desc="Calculating dense embeddings for dictionary",
+                )
+            else:
+                iterations = range(0, len(names), batch_size)
+
+            for start in iterations:
+
+                # make batch
+                end = min(start + batch_size, len(names))
+                batch = [Sentence(name) for name in names[start:end]]
+
+                # embed batch
+                self.encoder.embed(batch)
+
+                dense_embeds += [
+                    name.embedding.cpu().detach().numpy() for name in batch
+                ]
+
+                if flair.device.type == "cuda":
+                    torch.cuda.empty_cache()
```

Comment:
Is this necessary? I vaguely remember `empty_cache` to be an expensive operation.